fs/fs_zenfs.cc

```diff
@@ -481,6 +481,27 @@ std::shared_ptr<ZoneFile> ZenFS::GetFile(std::string fname) {
   return zoneFile;
 }
 
+/* Remove the link entry in ZoneFile and filename from the
+ * files_ map.
+ */
+IOStatus ZenFS::DeleteFileLink(std::string fname,
+                               std::shared_ptr<ZoneFile> zoneFile) {
+  IOStatus s;
+  std::string record;
+
+  files_.erase(fname);
+  zoneFile->RemoveLinkInfo(fname);
+  EncodeFileLinkDeletionTo(zoneFile, &record, fname);
+  s = PersistRecord(record);
+  if (!s.ok()) {
+    /* Failed to sync, return to a consistent state */
+    zoneFile->SetLinkInfo(fname.c_str());
+    zoneFile->SetNrLinks(zoneFile->GetNrLinks() + 1);
```

Comment:
Why is it necessary to increment the `nr_links_ one` more time? `SetLinkInfo` should take care of it or not?

Reply:
Yes, you are right. SetNrLinks()  is not needed here.